cmd: add config check for almost all commands

oleg-jukovec opened the pull request:
The check has been added for:

- binaries
- build
- check
- clean
- create
- install
- instances
- logrotate
- pack
- restart
- run
- start
- status
- stop
- uninstall

Closes #371

The patch does not affect:

-  cartridge
-  cat
-  cfg
-  completion
-  connect
-  coredump
-  daemon
-  help
-  init
-  play
-  rocks 
-  search
-  version

I'm not sure about the command list. I suggest to add commands in a review.

better0fdead commented on cli/cmd/common.go:
```diff
@@ -22,3 +23,13 @@ func handleCmdErr(cmd *cobra.Command, err error) {
+var errNoConfig = errors.New("tt.yaml not found, you need to create tt" +
+	" environment config with tt init")
+
+func checkConfig(cmdCtx *cmdcontext.CmdCtx) error {
```
i would like to see small description to function

oleg-jukovec replied:
Added.

better0fdead commented on cli/cmd/common.go:
```diff
@@ -22,3 +23,13 @@ func handleCmdErr(cmd *cobra.Command, err error) {
 }
+
+var errNoConfig = errors.New("tt.yaml not found, you need to create tt" +
+	" environment config with tt init")
```
```suggestion
	" environment config with 'tt init'")
```
up to you, i think it looks better to highlight commands in (').

oleg-jukovec replied:
Thank you, updated.

better0fdead commented:
Thx for patch, just small comments. I think commit title should be:
```cmd: add config check for almost commands```=>```cmd: add config check for almost all commands```

oleg-jukovec commented:
> `cmd: add config check for almost commands`=>`cmd: add config check for almost all commands`

Thank you, updated.

psergee commented on cli/cmd/common.go:
```diff
@@ -22,3 +23,15 @@ func handleCmdErr(cmd *cobra.Command, err error) {
 }
+
+// errNoConfig is returned if environment config file tt.yaml not found.
+var errNoConfig = errors.New("tt.yaml not found, you need to create tt" +
```
Let's use ConfigName from the `configure` package instead of `tt.yaml`.

oleg-jukovec replied:
Thank you, updated.

psergee commented on cli/cmd/common.go:
```diff
@@ -22,3 +23,15 @@ func handleCmdErr(cmd *cobra.Command, err error) {
+
+// errNoConfig is returned if environment config file tt.yaml not found.
+var errNoConfig = errors.New("tt.yaml not found, you need to create tt" +
+	" environment config with 'tt init'")
```
" or provide exact config location with `--cfg` option" :)

oleg-jukovec replied:
Added.

psergee commented on cli/cmd/common.go:
```diff
@@ -22,3 +23,15 @@ func handleCmdErr(cmd *cobra.Command, err error) {
+	" environment config with 'tt init'")
+
+// checkConfig returns an error if environment config file tt.yaml not found.
+func checkConfig(cmdCtx *cmdcontext.CmdCtx) error {
```
It is not clear from the function name what it does. Can we have something like "[is]ConfigFound" or "[is]ConfigFileLoaded" returning bool instead of error?

oleg-jukovec replied:
I've renamed `checkConfig` -> `checkConfigExist`.

It looks easier:

```Go
if err != checkConfigExist(ctx); err != nil {
    return err
}
```

Than:

```Go
if ok != isConfigExist(ctx); !ok {
    return someErr
}
```

for me. But I can redo it if you insist.

psergee replied:
What do you think about 
```go
if !configExists(ctx) {
    . . . 
}
```
?
Errors are mostly used for exceptional cases. Missing config does not look exceptional for `exists` kind of functions.
But this is for sure exceptional for start command implementation, for example.

psergee commented:
Suppose, we need to check for config in 
connect. to be able to connect using instances names
logrotate. command accepts app name. Where will it search for app log without config?

psergee commented:
- Update changelog.
- Add a comment with list of the commands which work without a config file.

oleg-jukovec commented:
> Suppose, we need to check for config in logrotate. command accepts app name. Where will it search for app log without config?

I've added the check for logrotate.

> Update changelog.

Done.

> Add a comment with list of the commands which work without a config file.

I've updated the pull request message. Did you mean to add a comment somewhere else?

psergee commented on CHANGELOG.md:
```diff
@@ -40,6 +40,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Do not use `make` command options for `tarantool` build if `MAKEFLAGS` environment variable
 is set.
+- `binaries`, `build`, `check`, `clean`, `create`, `install`, `instances`, `logrotate`, `pack`,
+  `restart`, `run`, `start`, `stop`, `uninstall` require environment configuration file.
```
```suggestion
  `restart`, `run`, `start`, `status`, `stop`, `uninstall` require environment configuration file.
```

oleg-jukovec replied:
Fixed.
